src/common/hooks/use-recently-viewed-tx.tsx

```diff
@@ -0,0 +1,21 @@
+import { useState, useEffect } from 'react';
+import { store } from '@common/utils';
+
+const storageKey = 'recentlyViewedTx';
+
+export const useRecentlyViewedTx = (txid?: string) => {
+  if (!process.browser) return;
+
+  const storedTxs = store.get(storageKey, []);
+  const [viewedTransactions, setViewedTransactions] = useState(storedTxs);
+
+  useEffect(() => {
+    if (txid) {
+      const newTxList = [...viewedTransactions, txid];
```

Comment:
Easy solution for now: 

```suggestion
      const newTxList = [...new Set([...viewedTransactions, txid])];
```